src/adapter/graphql/src/scalars/flow_configuration.rs

```diff
@@ -91,8 +91,13 @@ pub enum TimeUnit {
 
 impl From<chrono::Duration> for TimeDelta {
     fn from(value: chrono::Duration) -> Self {
+        if value.num_seconds() <= 0 {
+            panic!("Positive interval expected, but received [{value}]")
+        }
+
         let num_weeks = value.num_weeks();
         if (value - chrono::Duration::weeks(num_weeks)).is_zero() {
+            #[allow(clippy::cast_sign_loss, clippy::cast_possible_truncation)]
```

Comment:
Will the linting situation change, if we place assertions on the values instead of pragma?
This might indicate `Duration` is not the best type we could have used, as it can be negative too, while our model allows only positive values, and in limited range.